### PR TITLE
show procedure to fix missing HMN_MTN and NMN_MTN networks for csm pr…

### PR DESCRIPTION
**Pull request by jacobsalmela**
…ovider

# Summary and Scope

- adds docs for creating missing SLS networks
- links to docs from within the SLS validator

# Risks and Mitigations

### CODE REVIEW

File: docs/troubleshooting/known_errors.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+## `Required network HMN_MTN is missing` or `Required network NMN_MTN is missing`
+
+For the CSM provider, this means the existing SLS data is unstable as the `Networks` key is missing a required field.
+
+To fix the issue, craft an appropriate JSON file and use it to create the network in SLS.
+
+### Create An HMN_MTN or NMN_MTN Network JSON File
+
+Here is an example HMN_MTN JSON file that creates the network when patched into SLS. 
+> Note: update the site-specific information as needed (IP, MTU, VLANs, etc.)
```
Comment (rsjostrand-hpe) [inline docs/troubleshooting/known_errors.md:12]: The MTU value shouldn't be changed.

File: docs/troubleshooting/known_errors.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  "VlanRange": [
+    4094,
+    0
+  ]
+}
+```
+
+Here is the same, but for NMN_MTN.
+
+> Note: update the site-specific information as needed (IP, MTU, VLANs, etc.)
```
Comment (rsjostrand-hpe) [inline docs/troubleshooting/known_errors.md:36]: The MTU value shouldn't be changed.

File: docs/troubleshooting/known_errors.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+}
+```
+
+
+
+### Create The Networks in SLS
+
+Use the `cray` command to add the network(s):
+
+```shell
```
Comment (rsjostrand-hpe) [inline docs/troubleshooting/known_errors.md:64]: Let's consider just specify the CrayCLI commands.

If we keep the curl commands they are missing the authentication token.

I wonder if we need to cover the following scenarios, and split the commands for each scenario into a subsection and follow the one the applies.
* Cray CLI (the steps would be the same on and off the cluster)
* Curl when on a `ncn-*` node, targeting the normal/NMN API gateway
* Curl when on a remote system targeting the CMN API Gateway.

**Review by rsjostrand-hpe — CHANGES_REQUESTED** [review]

File: docs/troubleshooting/known_errors.md (hunk 2 above)
Comment (jacobsalmela) [reply docs/troubleshooting/known_errors.md]: Fixed

File: docs/troubleshooting/known_errors.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+## `Required network HMN_MTN is missing` or `Required network NMN_MTN is missing`
+
+For the CSM provider, this means the existing SLS data is unstable as the `Networks` key is missing a required field.
+
+To fix the issue, craft an appropriate JSON file and use it to create the network in SLS.
+
+### Create An HMN_MTN or NMN_MTN Network JSON File
+
+Here is an example HMN_MTN JSON file that creates the network when patched into SLS.
```
[the remark this answers is not in the document]
Comment (jacobsalmela) [reply docs/troubleshooting/known_errors.md]: Fixed

File: docs/troubleshooting/known_errors.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  "Type": "ethernet",
+  "ExtraProperties": {
+    "CIDR": "10.104.0.0/17",
+    "MTU": 9000,
+    "Subnets": []
+  },
+  "VlanRange": [
+    4094,
+    0
+  ]
```
Comment (rsjostrand-hpe) [inline docs/troubleshooting/known_errors.md:31]:
```
  "VlanRange": [
    2000,
    2999
  ]
```

File: docs/troubleshooting/known_errors.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  ],
+  "Type": "ethernet",
+  "ExtraProperties": {
+    "CIDR": "10.100.0.0/17",
+    "MTU": 9000,
+    "Subnets": [],
+    "VlanRange": [
+      4094,
+      0
+    ]
```
[the remark this answers is not in the document]
Comment (jacobsalmela) [reply docs/troubleshooting/known_errors.md]: Fixed

File: docs/troubleshooting/known_errors.md (hunk 5 above)
Comment (jacobsalmela) [reply docs/troubleshooting/known_errors.md]: Fixed

**Review by rsjostrand-hpe — APPROVED** [review]

**Pull request merged**